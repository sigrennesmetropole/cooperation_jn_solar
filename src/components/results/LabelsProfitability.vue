<script setup lang="ts">
import expandIcon from '@/assets/icons/expand-small-bigger-retract-smaller-big.svg'

const props = defineProps<{
  link: string
}>()
</script>

<template>
  <div class="flex flex-col gap-6 w-[100%] bg-slate-100 rounded-lg p-6 mx-auto">
    <slot name="img" class="rounded-xl"></slot>
    <slot name="title"></slot>
    <slot name="icon"></slot>
    <div class="flex flex-row">
      <p class="underline decoration-1 font-medium text-base">
        <a :href="props.link" target="_blank"> Accéder à la page </a>
      </p>
      <a :href="props.link" target="_blank">
        <img :src="expandIcon" class="w-3.5 h-3.5 ml-2 mt-1" />
      </a>
    </div>
  </div>
</template>
