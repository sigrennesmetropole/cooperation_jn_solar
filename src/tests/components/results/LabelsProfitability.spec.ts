import { mount } from '@vue/test-utils'
import LabelsProfitability from '@/components/results/LabelsProfitability.vue'

describe('LabelsProfitability', () => {
  const wrapper = mount(LabelsProfitability, {
    slots: {
      img: '<img src="test-image.jpg" alt="Test Image">',
      title: '<div>Title test</div>',
      icon: '<div></div>',
    },
    props: {
      link: 'https://www.photovoltaique.info/fr',
    },
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('displays link', () => {
    const photovoltaiqueLink = wrapper.find('p')
    expect(photovoltaiqueLink.exists()).toBe(true)
  })

  it('renders a link with the correct href', () => {
    const link = 'https://www.photovoltaique.info/fr/'
    const wrapper = mount(LabelsProfitability, {
      props: { link },
      slots: {
        img: '<img src="test-image.jpg" alt="Test Image">',
        title: '<div>Title test</div>',
        icon: '<div></div>',
      },
      global: {
        stubs: {
          UiLinkWithIconArrow: {
            template: '<a :href="link" target="_blank" class="stub"></a>',
            props: ['link'],
          },
        },
      },
    })

    const a = wrapper.find('.stub')
    expect(a.attributes('href')).toBe(link)
  })
})
