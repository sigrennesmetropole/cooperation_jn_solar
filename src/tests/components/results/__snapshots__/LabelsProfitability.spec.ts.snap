// Vitest Snapshot v1

exports[`LabelsProfitability > renders correctly 1`] = `
"<div class=\\"flex flex-col gap-6 w-[100%] bg-slate-100 rounded-lg p-6 mx-auto\\"><img src=\\"test-image.jpg\\" alt=\\"Test Image\\">
  <div>Title test</div>
  <div></div>
  <div class=\\"flex flex-row cursor-pointer w-fit\\" tabindex=\\"0\\">
    <p class=\\"underline decoration-1 font-medium text-base\\"> Accéder à la page </p><img src=\\"/src/assets/icons/expand-small-bigger-retract-smaller-big.svg\\" class=\\"w-3.5 h-3.5 ml-2 mt-1\\">
  </div>
</div>"
`;
